frontend/src/routes/__root.tsx

```diff
@@ -0,0 +1,18 @@
+import { createRootRoute, Link, Outlet } from "@tanstack/react-router";
+
+export const Route = createRootRoute({
+  component: () => (
+    <>
+      <div className="p-2 flex gap-2">
+        <Link to="/" className="[&.active]:font-bold">
+          Home
+        </Link>{" "}
+        <Link to="/login" className="[&.active]:font-bold">
```

Comment:
Why  use `[&.active]` instead of just `font-bold` ?

Reply:
[&.active] to help the toogling

Reply:
Oh... tanstack router adds an `active` class name to active links ? 🤔

Reply:
Yup. you got it right

Reply:
Can you try to maybe wrap the `Link` in a `<NavLink>` component that adds the `data-active` prop instead of the class name ?

You can take some inspiration from here:  https://tanstack.com/router/v1/docs/framework/react/guide/navigation#link-component